### PR TITLE
Fix : Remove all the warning to compile the code without any.

### LINKED ISSUE #67: Too many warnings
### Describe the bug
The applications runs with too many warnings.

### To Reproduce
1. Go to each and every file where the warning is given
2. Go on that line where the warning is given.
3. See the warning the resolve that.

### Expected behavior
After resolving all the warnings we get to run the applications without any warnings
### Screenshots
Screenshot 2021-01-28 at 8 36 58 AM

### Desktop (please complete the following information):
 - OS: iOS
 - Browser: safari, chrome
 - Version: 22

### PR DESCRIPTION
**Pull request by sagar-barapatre**
### Description

There were a total of 16 warnings when the code was compiled locally. The main purpose of this PR was to remove all the warnings before hand so that the code compiles without any such warnings.

Fixes #67 

### Type of Change: bug

- Code
- Quality Assurance
- Outreach

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Screenshot 2021-01-28 at 9 21 43 AM

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] I have written Kotlin Docs whenever is applicable

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by isabelcosta** [comment]
@sagar-barapatre can you please fix the merge conflicts here?

**Comment by sagar-barapatre** [comment]
@isabelcosta I have fixed the conflicts. Can you please have a look.

**Comment by Amulya-coder** [comment]
@sagar-barapatre please squash your commits into one. It's always better to have one commit per change to keep the history clean.

**Comment by khushishikhu** [comment]
@sagar-barapatre  You should change the PR title according to AnitaB.org's guidelines

**Review by khushishikhu — COMMENTED** [review]
@sagar-barapatre You should change the PR title according to the AnitaB.org's guidelines and also the commit message style

**Comment by sagar-barapatre** [comment]
@khushishikhu I have changed the commit message style but I am unable to figure out what should be the proper PR title. I didn't found it in the [Contributing Guidelines](https://github.com/anitab-org/open-source-programs-web/blob/develop/.github/PULL_REQUEST_TEMPLATE.md)

**Comment by jalajcodes** [comment]
> @khushishikhu I have changed the commit message style but I am unable to figure out what should be the proper PR title. I didn't found it in the [Contributing Guidelines](https://github.com/anitab-org/open-source-programs-web/blob/develop/.github/PULL_REQUEST_TEMPLATE.md)

Adding a prefix just like the commit messages should do it.

**Comment by sagar-barapatre** [comment]
@isabelcosta Please have a look now !

File: src/components/Login.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import login from './../styles/Login.css'
-import orgLogo from '../assets/org-logo.jpg'
-import { register } from '../urls'
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { postLogin } from '../actions/login';
+import { Form, Grid, Image, Divider, Icon, Message, Button } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+// import login from './../styles/Login.css'
```
Comment (isabelcosta) [inline src/components/Login.js:7]: is this not needed?

File: src/components/Profile.js [hunk 2]
```diff
@@ -59,9 +59,18 @@ class Profile extends Component {
...
+        if (userinfo && userinfo.length === 1) {
+            this.setState({
+                update : true
+            })
+        } else {
+            this.setState({
+                update: false
+            })
         }
+            // userinfo && userinfo.length === 1 ? this.state.update = true : this.state.update = false
```
Comment (isabelcosta) [inline src/components/Profile.js:72]: what is this lane about 🤔

File: src/components/Navbar.js [hunk 3]
```diff
@@ -42,6 +43,11 @@ class Navbar extends Component {
         const result = this.props.userinfo
         const { user } = this.props
         const { isLoggedIn } = this.state
+       
+
+//    let loginButton = localStorage.getItem('token') !== null? this.state.isLoggedIn = true: this.state.isLoggedIn = false;
+//    this.setState({loginButton: loginButton});
```
Comment (isabelcosta) [inline src/components/Navbar.js]: should this be here?

File: src/components/Login.js (hunk 1 above)
Comment (sagar-barapatre) [reply src/components/Login.js:7]: The import login wasn't used anywhere in Login.js file which resulted in the warning "Used of undeclared variable login".
That's why we need to comment out that.

File: src/components/Profile.js (hunk 2 above)
Comment (sagar-barapatre) [reply src/components/Profile.js:72]: This warning was like we have to use setState() to mutate the props so I commented out this code and use the setState() to remove the warning

File: src/components/Navbar.js (hunk 3 above)
Comment (sagar-barapatre) [reply src/components/Navbar.js]: This is also the same, warning was to use setState() instead of props to mutate it. So commenting that piece of code and writing the new one did workout for me.

Comment (isabelcosta) [reply src/components/Navbar.js]: Ok so you re-implemented the commented code? if so, then you can remove the commented code @sagar-barapatre

Comment (sagar-barapatre) [reply src/components/Navbar.js]: @isabelcosta Yeah done !!

File: src/components/Profile.js (hunk 2 above)
Comment (jalajcodes) [reply src/components/Profile.js:72]: @sagar-barapatre I don't think its a good idea to use `setState` inside `render`. It [not advisable](https://stackoverflow.com/a/48226704/5182477)

This code should be modified in a way so that we don't have to mutate state inside render. For now I think you should revert the changes and disable eslint to supress the warnings. We can create another issue to fix it.

File: src/components/Login.js (hunk 1 above)
Comment (jalajcodes) [reply src/components/Login.js:7]: Have you tried rewriting it like this `import './../styles/Login.css'` ?

File: src/components/Menubar.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-import { dashboard, forms, submission } from '../urls'
-import menubar from './../styles/Menubar.css'
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getInfo } from '../actions/info';
+import PropTypes from 'prop-types';
+import { Menu } from 'semantic-ui-react';
+import { dashboard, forms, submission } from '../urls';
+// import menubar from './../styles/Menubar.css'
```
Comment (jalajcodes) [inline src/components/Menubar.js:8]: same here

File: src/store.js [hunk 5]
```diff
@@ -1,14 +1,11 @@
 import { createBrowserHistory } from 'history'
-import { apiMiddleware } from 'redux-api-middleware';
 import { applyMiddleware, createStore, compose } from 'redux'
 import { routerMiddleware } from 'react-router-redux'
 import thunk from 'redux-thunk'
 import rootReducers from './reducers'
 
 export const history = createBrowserHistory();
 
-const createStoreWithMiddleware = applyMiddleware(apiMiddleware)(createStore);
```
Comment (jalajcodes) [inline src/store.js:10]: why did you remove that line?

**Review by jalajcodes — COMMENTED** [review]
Please checkout my comments below

**Comment by Rahulm2310** [comment]
@sagar-barapatre Please don't use `develop` branch for submitting a pull request. Always use a seperate/new branch. It would be great if you could close this PR and open a new PR from a seperate branch. Thanks 🙂   

Cc: @isabelcosta @codesankalp

File: src/components/Login.js (hunk 1 above)
Comment (sagar-barapatre) [reply src/components/Login.js:7]: No

File: src/components/Profile.js (hunk 2 above)
Comment (sagar-barapatre) [reply src/components/Profile.js:72]: In warning it was specifically mentioned to use setState() to remove that

File: src/components/Menubar.js (hunk 4 above)
Comment (sagar-barapatre) [reply src/components/Menubar.js:8]: No, I just commented out that code because it was never used in that file.

File: src/store.js (hunk 5 above)
Comment (sagar-barapatre) [reply src/store.js:10]: because "createStoreWithMiddleware" was declared here and this variable was also never used in that file. So unnecessary declaration of variable was causing error.

**Comment by codesankalp** [comment]
Also, I think all the warnings have been resolved in this PR https://github.com/anitab-org/open-source-programs-web/pull/94 @isabelcosta. So should I close this?

**Comment by codesankalp** [comment]
Closing this PR as it is fixed after adding QA-Checks and linting.
